flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -881,24 +883,21 @@ class GroupWindowedTable(
     * }}}
     */
   def select(fields: Expression*): Table = {
```

Comment:
What if user use a customized watermark extracter which used some fields from the element. For example, we have original table source containing 4 fields: a, b, c, d. And user used "a" field to extract the timestamp and watermark. But in the later query on the table, only "b" and "c" are used. And if we do a projection on "b" and "c", and the projection is pushed into table source further, we will not get field "a" which used to produce timestamp anymore.